# filepath: /balance_controller_ros2/balance_controller_ros2/config/params.yaml
balance_controller:
  odrive:
    port: /dev/ttyUSB0
    baud: 115200
    axis_drive_index: 0
    axis_flywheel_index: 1

  pid:
    angle_kp: 350.0
    angle_ki: 0.0
    angle_kd: 0.7
    angle_v_kp: 0.45
    angle_v_ki: 0.0
    angle_v_kd: 0.1
    flywheel_speed_kp: 0.1
    flywheel_speed_ki: 0.45
    flywheel_speed_integral_limit:
      - -400.0
      - 400.0

  control:
    machine_middle_angle_init: -1.77
    bike_turn_scale_deg: 0.057
    bike_speed_scale_deg: 0.19
    middle_angle_recitfy_limit_deg: 3.0
    roll_diff_limit: 0.1
    march_velocity: 0.4
    drive_accel: 0.1
    flywheel_speed_limit: 20.0
    flywheel_accel_limit: 15.0

  servo:
    center_deg: 8.0
    range_each_side_deg: 20.0
    speed_dps: 2.0
    step_time_ms: 1000
    middle_range_deg: 2.0
    step_deg: 10.0

  safety:
    person_safety_distance: 1.5
    obstacle_safety_distance: 1.0
    visualize: false
    tacle_safety_distance: 1.0